docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,511 @@
+
+    return createStore(reducer, initialState, middlewares);
+    }
+}
+
+export default buildStore;
+```
+
+Then, in your actions scripts, you can use the provided client to fetch the content that you need, like so:
+
+```js
+// usecases/actions.js
+// Create a fetch action that fetches data from Contentful
+const fetch = () => async (dispatch, getState, { client }) => {
+    dispatch({ type: 'Fetching'})
+
+    try {
+        // Fetch content that are instances of `useCase` content model, in Spanish
+        const result = await client.getEntries({
+            content_type: 'useCase',
+            locale: 'es-ES',
+        })
+
+        // The incoming object will be fairly complex, and it's recommended to build
+        // a more conveniently structured object here so that other scrips in the pipeline
+        // can avoid handling that complexity and can instead enjoy a simpler, custom built object.
+        // E.G.
+        const entries = result.items
+            .map((item)=> ({
+                title: item.fields.title,
+                description: item.fields.description,
+                weight: item.fields.weight,
+        })).sort((a, b) => b.weight - a.weight);
+
+        dispatch({
+            type: 'Success',
+            payload: { entries },
+        })
+    } catch (error) {
+        dispatch({
+            type: 'Failure',
+            payload: { error },
+        });
+    }
+};
+```
+
+### 3. Conditionally rendering DOM elements in case of preview
+
+Since we're on the topic, you must use a similar strategy in your `App.js` file to conditionally render a DOM element to let the user know they are in a preview state. This must happen once per instance of the App, since we will not perpetuate the `cms-preview` query parameter on route changes.
+
+Class component:
+
+```js
+// App.js
+class App extends Nextapp {
+    state = {
+        isPreviewingContentful = false,
+    };
+
+    // Using the componentWillMount() lifecycle function guarantees that this runs only once per instance of App
+    componentWillMount() {
+        const { router } = this.props;
+
+        this.setState({
+            isPreviewingContentful: Object.hasOwnProperty.call(router.query, 'cms-preview');
+        });
+    }
+
+    render() {
+        return (
+            (...)
+                { this.isPreviewingContentful && <ContentfulPreview> }
+            (...)
+        );
+    }
+}
+```
+
+Using hooks: 
+
+```js
+// App.js
+const App = ({ Component, pageProps, rootSelector, router }) => {
+    const [isPreviewingContentful, setIsPreviewingContentful] = useState(false);
+    
+    // Using a useEffect hook with no dependencies guaranties that it fires only once per instance of App
+    useEffect(() => setIsPreviewingContentful(Object.hasOwnProperty.call(router.query, 'cms-preview')), []);
+
+    return (
+        (...)
+            { this.isPreviewingContentful && <ContentfulPreview> }
```

Comment:
`this` Is not valid here.

Missing / in the JSX tag of ContentfulPreview. Maybe we should add a note stating that it’s a component that would render a visible warning on the site saying we are on preview mode, like a ribbon. This is valid for the dato CMS recipe as well.